pkg/registration/spoke/bootstrapkubeconfigsmanager/bootstrapkubeconfig.go

```diff
@@ -0,0 +1,99 @@
+package bootstrapkubeconfigsmanager
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type boostrapKubeConfigStatus string
+
+const (
+	boostrapKubeConfigStatusInValid boostrapKubeConfigStatus = "InValid"
+	boostrapKubeConfigStatusValid   boostrapKubeConfigStatus = "Valid"
+)
+
+// bootstrapKubeConfig represents a bootstrap kubeconfig that agent can use to bootstrap a managed cluster.
+type boostrapKubeConfig interface {
+	// Name returns the name of the bootstrap kubeconfig. It helps to identify the bootstrap kubeconfig.
+	Name() string
+
+	// KubeConfigData includes the kubeconfig and the credentials to connect to the hub cluster.
+	KubeConfigData() (map[string][]byte, error)
+
+	// Status returns the status of the bootstrap kubeconfig.
+	// A bootstrap kubeconfig has two status: Valid and InValid.
+	Status() (boostrapKubeConfigStatus, error)
+
+	// Fail means at the time t, the bootstrap kubeconfig failed to connect to the hub cluster.
+	Fail(t time.Time) error
```

Comment:
what is the purpose of this method? When it returns error

Reply:
Change to MarkFail. And remove input t.

Reply:
Done.